appcues/src/main/java/com/appcues/action/appcues/StepInteractionAction.kt

```diff
@@ -0,0 +1,57 @@
+package com.appcues.action.appcues
+
+import com.appcues.Appcues
+import com.appcues.action.ExperienceAction
+import com.appcues.analytics.AnalyticsTracker
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType
+import com.appcues.data.model.AppcuesConfigMap
+import com.appcues.statemachine.StateMachine
+
+internal class StepInteractionAction(
+    override val config: AppcuesConfigMap,
+    private val interaction: StepInteractionData,
+    private val analyticsTracker: AnalyticsTracker,
+    private val stateMachine: StateMachine,
+) : ExperienceAction {
+
+    class StepInteractionData(
```

Comment:
this class represents the data analytics will merge to the step_interaction track event.